packages/plugin-ext/src/main/browser/webviews-main.ts

```diff
@@ -144,10 +216,34 @@ export class WebviewsMainImpl implements WebviewsMain {
         return Promise.resolve(webview !== undefined);
     }
     $registerSerializer(viewType: string): void {
-        throw new Error('Method not implemented.');
+        this.revivers.add(viewType);
     }
     $unregisterSerializer(viewType: string): void {
-        throw new Error('Method not implemented.');
+        this.revivers.delete(viewType);
+    }
+
+    private async checkViewOptions(handler: string, viewColumn?: number | undefined) {
+        const options = this.viewsOptions.get(handler);
+        if (!options || !options.panelOptions) {
+            return;
+        }
+        const view = this.views.get(options.panelId);
+        if (!view) {
+            return;
+        }
+        const active = !!this.shell.activeWidget ? this.shell.activeWidget.id === view!.id : false;
+        const visible = view!.isVisible;
+        if (viewColumn === undefined) {
+            this.viewColumnService.updateViewColumns();
+            viewColumn = this.viewColumnService.hasViewColumn(view.id) ? this.viewColumnService.getViewColumn(view.id)! : 0;
+        }
+        if (viewColumn === undefined && options.panelOptions.viewColumn === viewColumn && options.visible === visible && options.active === active) {
```

Comment:
Is it a check to prevent firing an event if state did not change?

If it is so `viewColumn === undefined && options.panelOptions.viewColumn === viewColumn` looks strange. Did you mean `viewColumn === undefined || (...)` ?

Reply:
fixed